lightning-invoice/src/lib.rs

```diff
@@ -1415,6 +1415,15 @@ impl Deref for SignedRawInvoice {
 	}
 }
 
+impl std::hash::Hash for InvoiceSignature {
+	fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+		let (recovery_id, sig) = self.0.serialize_compact();
+		let recovery_id = recovery_id.to_i32();
+		state.write(&sig);
+		state.write_i32(recovery_id);
```

Comment:
I'm not sure if it is ok to do it like this. In secp256k1 they already have a Hash impl for the contained type in ffi `pub struct RecoverableSignature(ffi::RecoverableSignature);` but they don't derive Hash on that. If ffi was public the correct way would be `(&self.0.0[..]).hash(state)` but it is not possible